include/core/jbackend.h

```diff
@@ -61,79 +66,365 @@ struct JBackend
+@param name [in] schema name to delete (e.g. "files")
+@param selector [in] the selector to decide which data should be updated. points to
+ - an initialized bson containing "selector_part_and"
+ - an empty bson
+ - NULL
+@verbatim
+selector_part_and: {
+	"var_name1": {
+		"operator": op1 (int32),
+		"value": value1,
+	},
+	"var_nameN": {
+		"operator": op2 (int32),
+		"value": value2,
+	}
+	_or: bson_t(selector_part_or)
+}
+selector_part_or: {
+	"var_name1": {
+		"operator": op (int32),
+		"value": value,
+	},
+	"var_nameN": {
+		"operator": op (int32),
+		"value": value,
+	}
+	_and: bson_t(selector_part_and)
+}
+@endverbatim
+@return TRUE if all following statements are TRUE otherwise FALSE
+	- (namespace, name) did exists before
+	- there are no var_names which are not existent in the schema definition
+	- there is at least one element deleted in the db-backend
+*/
+			gboolean (*backend_delete)(gpointer batch, gchar const* name, bson_t const* selector, GError** error);
+			/*!
+creates an iterator for the db-backend
+@param namespace [in] different usecases (e.g. "adios", "hdf5")
+@param name [in] schema name to delete (e.g. "files")
+@param selector [in] the selector to decide which data should be updated. points to
+ - an initialized bson containing "selector_part_and"
+ - an empty bson
+ - NULL
+@verbatim
+selector_part_and: {
+	"var_name1": {
+		"operator": op1 (int32),
+		"value": value1,
+	},
+	"var_nameN": {
+		"operator": op2 (int32),
+		"value": value2,
+	}
+	_or: bson_t(selector_part_or)
+}
+selector_part_or: {
+	"var_name1": {
+		"operator": op (int32),
+		"value": value,
+	},
+	"var_nameN": {
+		"operator": op (int32),
+		"value": value,
+	}
+	_and: bson_t(selector_part_and)
+}
+@endverbatim
+@param iterator [out] the iterator which can be used later for backend_iterate
+@return TRUE if all following statements are TRUE otherwise FALSE
+	- (namespace, name) did exists before
+	- there are no var_names which are not existent in the schema definition
+	- the iterator is valid
+*/
+			gboolean (*backend_query)(gpointer batch, gchar const* name, bson_t const* selector, gpointer* iterator, GError** error);
+			/*!
+obtains metadata from the backend
+backend_iterate should be called until the returned value is NULL due to no more elements found - this allowes the backend to free potentially 
+allocated caches.
+@param iterator [inout] the iterator specifying the data to retrieve
+@param metadata [out] the requested metadata initially points to
+ - an initialized bson - ready to write a document
+@verbatim
+{
+	"_id": value0,
+	"var_name1": value1,
+	"var_name2": value2,
+	"var_nameN": valueN,
+}
+@endverbatim
+@return TRUE if all following statements are TRUE otherwise FALSE
+	- (namespace, name) did exists before
+	- metadata was not NULL
+	- iterator pointed to an existing element
+	- metadata contains requested element fround by iterator
+	- iterator is updated to point to next element (if next element exists)
+*/
```

Comment:
The comments should probably be indented.

Reply:
I could indent everything, but the part with @verbatim includes spaces in the generated Doxygen-html. If I indent the comments in the code, they are not well aligned in the html.